apps/script/HoC2023ScriptFiles/README.md

```diff
@@ -1,25 +1,62 @@
 An embedding is a vector (list) of floating point numbers, and embeddings are used to
 measure the relatedness of text strings. The distance between two vectors measures
-their relatedness. https://platform.openai.com/docs/guides/embeddings
-We also store embeddings in caches files as as pickle files, python's native way to serialize data.
+their relatedness. https://developers.google.com/machine-learning/crash-course/embeddings/video-lecture
+These embeddings are stored in caches files as pickle files, python's native way to serialize data.
+
+## Architecture and Background
+
+At runtime, DanceAI will use the three maps to lookup the scores for each output type and take the top three indexes of (MAX(SUM(Input1Scores, Input2Scores, Input3Scores))) to select a final palette/foreground/background to display to the user. These maps are stored as a local cache rather than generated at runtime to remove the costs associated with querying a LLM and improve runtime performance.
+
+The maps themselves are generated with the following pieces of intuition and underlying assumptions:
+
+1. We can approximate the associated "mood" of an abstract concept (e.g. the relationship between Fire <-> Warm Color Palette) by calculating the similarity/co-occurrence of the pair together in text
+2. LLMs trained on sufficiently large amounts of data should be able to provide embeddings for #1
+3. Each input should only have a strong association with a small number of outputs rather than all outputs
+
+To aid in ensuring point #3 occurs, many of the inputs and possible outputs have been renamed from their base unicode or blockly_id identifiers to try to provide more feature rich information for the embedding that the LLM will produce. These names are both listed below and stored within constants.py and apps/static/dance/ai/ai-inputs.json for outputs and emoji inputs respectively.
+
+A simple flow diagram of how data/parameters move through the script would be:
+(constants.py + ai-inputs.json) -> HoC2023AiGenerateWeights.py -> Hoc2023AiHelperFunctions.py -> [LLM of Choice] -> HoC2023AiGenerateWeights.py -> Final Map Output
+
+## AWS Titan Setup and Quirks
+
+To be blunt, setting up AWS is challenging and the guidance/code samples change frequently. It is highly suggested to check their documentation at https://docs.aws.amazon.com/bedrock/latest/userguide/model-access.html, https://github.com/aws-samples/amazon-bedrock-workshop, and https://github.com/aws-samples/amazon-bedrock-workshop/blob/main/00_Intro/bedrock_boto3_setup.ipynb and do your best to follow along.
+
+Additionally, make sure to reach out to the Infrastructure team for best practices related to IAM permissions and role creation you will need to get Bedrock model access. Previously, we opted to utilize the recommended SageMaker service and role to streamline permissions issues but better options (for both cost and organization) might have since been determined.
+
+## Output Evaluation
```

Comment:
Probably better to have this in a dev doc and keep the README a bit more focused as a "how to" guide.

Reply:
Here's an example of a `README` that links to a dev doc in `@code-dot-org/javabuilder`: https://github.com/code-dot-org/javabuilder/blob/main/README.md